opengever/maintenance/scripts/fix_retention_period.py

```diff
@@ -0,0 +1,465 @@
+    def _diff(self):
+        self.reference_number = self._get_repository_reference_number()
+        if self.reference_number != self.item['reference_number']:
+            logger.warn('reference numbers differ for: {}'
+                        ' "{}" (site), "{}" (excel)'
+                        .format(self.item['_query_path'],
+                                self.reference_number,
+                                self.item["reference_number"]))
+            self.can_apply = False
+            return
+
+        self.current_period = ILifeCycle(self.context).retention_period
+        xls_period = self.item.get('retention_period')
+        self.new_period = xls_period or self.parent.new_period
+
+    def _get_repository_reference_number(self):
+        reference = IReferenceNumber(self.context)
+        formatter = getAdapter(self.context, IReferenceNumberFormatter,
+                               name=self.reference_formatter)
+        return formatter.repository_number(reference.get_parent_numbers())
+
+    def make_retention_period_backup(self, obj, backup_period):
+        annotations = IAnnotations(obj)
+        if BACKUP_KEY not in annotations:
+            annotations[BACKUP_KEY] = backup_period
+
+    def apply_recursively(self):
+        if not self.can_apply:
+            return
+
+        if self.apply_to_repo_folder() and self.is_leaf_folder:
+            self.apply_to_dossiers()
+
+        super(RepoFolderDiff, self).apply_recursively()
+
+    def apply_to_dossiers(self):
+        child_dossiers = self.catalog.unrestrictedSearchResults(
+            path='/'.join(self.context.getPhysicalPath()),
+            object_provides=IDossierMarker.__identifier__)
+
+        for brain in child_dossiers:
+            self.apply_to_dossier(brain.getObject())
+
+    def apply_to_repo_folder(self):
+        kind = 'leaf ' if self.is_leaf_folder else ''
+        if self.current_period != DEFAULT_PERIOD:
+            if self.options.verbose:
+                logger.info('skipping {}repo-folder ({}) {}, modified'
+                            .format(kind,
+                                    self.reference_number,
+                                    self.item['_query_path']))
+            return False
+
+        current_title = self.context.Title(prefix_with_reference_number=False)
+        current_title = current_title.decode('utf-8')  # Title returns utf-8
+        xls_title = self.item['effective_title']
+        if current_title != xls_title:
+            if self.options.verbose:
+                logger.info(u'skipping {}repo-folder ({}) {}, title changed '
+                            u'from "{}" to "{}"'
+                            .format(kind,
+                                    self.reference_number,
+                                    self.item['_query_path'],
+                                    xls_title,
+                                    current_title))
+            return False
+
+        self.fixed_folders.add(self.context)
```

Comment:
Marking it fixed before actually fixing it - folder may still be skipped below (albeit without any real effect). But I'd rather move `self.fixed_folders.add(self.context)` right before `self.add_journal_entry()` below. Or is the meaning of `fixed` rather something like "handled" / "considered" here?